element_array_ephys/ephys_precluster.py

```diff
@@ -0,0 +1,896 @@
+                    'Processing for neuropixels'
+                    ' probe model {} not yet implemented'.format(probe_data.probe_model))
+
+            self.insert1({**key,
+                          **generate_electrode_config(probe_type, electrode_group_members),
+                          'acq_software': acq_software,
+                          'sampling_rate': probe_data.ap_meta['sample_rate'],
+                          'recording_datetime': probe_data.recording_info['recording_datetimes'][0],
+                          'recording_duration': np.sum(probe_data.recording_info['recording_durations'])})
+
+            root_dir = find_root_directory(get_ephys_root_data_dir(),
+                probe_data.recording_info['recording_files'][0])
+            self.EphysFile.insert([{**key,
+                                    'file_path': fp.relative_to(root_dir).as_posix()}
+                                   for fp in probe_data.recording_info['recording_files']])
+        else:
+            raise NotImplementedError(f'Processing ephys files from'
+                                      f' acquisition software of type {acq_software} is'
+                                      f' not yet implemented')
+
+
+@schema
+class PreClusterMethod(dj.Lookup):
+    definition = """
+    # Method for pre-clustering
+    precluster_method: varchar(16)
+    ---
+    precluster_method_desc: varchar(1000)
+    """
+
+    contents = [('catgt', 'Time shift, Common average referencing, Zeroing')]
+
+
+@schema
+class PreClusterParamSet(dj.Lookup):
+    definition = """
+    # Parameter set to be used in a clustering procedure
+    paramset_idx:  smallint
+    ---
+    -> PreClusterMethod    
+    paramset_desc: varchar(128)
+    param_set_hash: uuid
+    unique index (param_set_hash)
+    params: longblob  # dictionary of all applicable parameters
+    """
+
+    @classmethod
+    def insert_new_params(cls, precluster_method: str, paramset_idx: int,
+                          paramset_desc: str, params: dict):
+        param_dict = {'precluster_method': precluster_method,
+                      'paramset_idx': paramset_idx,
+                      'paramset_desc': paramset_desc,
+                      'params': params,
+                      'param_set_hash':  dict_to_uuid(params)}
+        param_query = cls & {'param_set_hash': param_dict['param_set_hash']}
+
+        if param_query:  # If the specified param-set already exists
+            existing_paramset_idx = param_query.fetch1('paramset_idx')
+            if existing_paramset_idx == paramset_idx:  # If the existing set has the same paramset_idx: job done
+                return
+            else:  # If not same name: human error, trying to add the same paramset with different name
+                raise dj.DataJointError(
+                    'The specified param-set'
+                    ' already exists - paramset_idx: {}'.format(existing_paramset_idx))
+        else:
+            cls.insert1(param_dict)
```

Comment:
```suggestion
        cls.insert1(param_dict)
```

You do not need to explicitly handle the duplication errors. DataJoint will already throw a reasonable exception.

Reply:
If the datajoint exception is not clear enough, we can work to improve it.